crates/compiler/parse/src/expr.rs

```diff
@@ -222,6 +222,7 @@ fn loc_term_or_underscore<'a>(
         loc(underscore_expression()),
         loc(record_literal_help()),
         loc(specialize_err(EExpr::List, list_literal_help())),
+        loc(apply_with_pnc()),
```

Comment:
I think this is slightly the wrong layer to inject this at. Concretely, putting this here will mean that `dbg("test")`, and `crash("foo")` won't be parsed as PNC applies, which I think we want them to be. Similarly for parenthesized funcs - e.g. `(foo)(1, 2, 3)`.

I would somewhat pedantically say that e.g. `{foo:bar}(1, 2, 3)` and `[foo](1, 2, 3)`, `"foo"(1, 2, 3)`, etc - should all parse as a PNC apply, since that should hopefully let us give better error messages

Specifically, I would:
* Bite the bullet and allow `_` in expressions, allowing us to merge `loc_term_or_underscore` and `loc_term`.
* Extract a function from this newly merged thing, which consists of every thing in this list that could syntactically support being a function: at the very least: crash, dbg, and ident chains - but I'd suggest even including everything except closures from this current `loc_term_or_underscore` function - I'm going to arbitrarily call that new parser a "factor", but feel free to use whatever name speaks to you
* `loc_term_or_underscore` then becomes a oneof between:
  * closure
  * a factor, followed by zero or more parenthesized argument lists (we need the "more" to support manually curried function calls!)

Reply:
BTW by "allow _ in expressions", I just mean in the parser. Canonicalization should of course continue to reject it.

Reply:
To be clear on this, I'm really just throwing out `loc_term` and then putting everything but closure into a new parser that is then put into a one_of! with closure in `loc_term_or_underscore` and replace all usages of `loc_term` with `loc_term_or_underscore`?

Reply:
That sounds right, yes.